syntax = "proto3";

package ui;

service UI {
    rpc Ping(PingRequest) returns (PingReply) {}
    rpc AskRule (RuleRequest) returns (RuleReply) {}
}

message Statistics {
    uint64 uptime = 1;
	uint64 dns_responses = 2;
	uint64 connections  = 3;
	uint64 ignored = 4;
	uint64 accepted = 5;
	uint64 dropped = 6;
	uint64 rule_hits = 7;
	uint64 rule_misses = 8;
	map<string, uint64> by_proto = 9;
	map<string, uint64> by_address = 10;
	map<string, uint64> by_host = 11;
	map<string, uint64> by_port = 12;
	map<string, uint64> by_uid = 13;
	map<string, uint64> by_executable = 14;
}

message PingRequest {
    uint64 id = 1;
    Statistics stats = 2;
}

message PingReply {
    uint64 id = 1;
}

message RuleRequest {
    string protocol = 1;
    string src_ip = 2;
    uint32 src_port = 3;
    string dst_ip = 4;
    string dst_host = 5;
    uint32 dst_port = 6;
    uint32 user_id = 7;
    uint32 process_id = 8;
    string process_path = 9;
    repeated string process_args = 10;
}

message RuleReply {
    string name = 1;
    string action = 2;
    string duration = 3;
    string what = 4;
    string value = 5;
}
